backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -490,31 +558,31 @@ export class FeatureFlagService {
       });
     });
 
-    if (segmentValidator) {
-      return {
-        fileName: fileName,
-        compatibilityType: FF_COMPATIBILITY_TYPE.INCOMPATIBLE,
-      };
-    }
-
-    if (!flag.name || !flag.key || !flag.context) {
+    // check for missing fields
+    if (!flag.name || !flag.key || !flag.context || segmentValidator) {
       compatibilityType = FF_COMPATIBILITY_TYPE.INCOMPATIBLE;
     } else {
-      const segmentIds = [
-        ...flag.featureFlagSegmentInclusion.flatMap((segmentInclusion) =>
-          segmentInclusion.segment.subSegments.map((subSegment) => subSegment.id)
-        ),
-        ...flag.featureFlagSegmentExclusion.flatMap((segmentExclusion) =>
-          segmentExclusion.segment.subSegments.map((subSegment) => subSegment.id)
-        ),
-      ];
-
-      const segments = await this.segmentService.getSegmentByIds(segmentIds);
-      segments.forEach((segment) => {
-        if (segment == undefined) {
-          compatibilityType = FF_COMPATIBILITY_TYPE.WARNING;
-        }
-      });
+      const keyExists = await this.featureFlagRepository.findOne({ key: flag.key });
```

Comment:
This change is from another PR. Right?

Reply:
Instead of doing `findOne` request for each flag, can we move the fetching in the calling function (find In array query) and send the result as the argument of `validateImportFeatureFlag`?

Reply:
code updated